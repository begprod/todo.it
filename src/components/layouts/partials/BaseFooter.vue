<template>
  <footer class="p-5 border-t bg-white">
    <div class="flex flex-col items-center justify-center text-sm text-gray-500">
      <p>
        made with <span class="text-red-500">❤</span> by
       <a href="https://github.com/begprod" class="text-blue-600">begprod</a>
      </p>
      <p>v.version</p>
      <p>
        <a
          class="text-blue-600"
          href="#"
          target="_blank"
          rel="noreferrer"
        >
          Install app on device instructions
          <span class="inline-block ml-2 -rotate-12 origin-center">→</span>
        </a>
      </p>
    </div>
  </footer>
</template>
