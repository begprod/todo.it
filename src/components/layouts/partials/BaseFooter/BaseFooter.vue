<template>
  <footer class="footer">
    <div>
      Crafted with <Heart class="footer__heart icon icon_base" /> by
      <br />
      <a href="https://github.com/begprod" class="link link_underline">Andrej Aratov</a>
    </div>

    <a
      class="link link_underline"
      href="https://github.com/begprod/todo.it#install-application-on-your-device"
      target="_blank"
      rel="noreferrer"
    >
      Install app on device instruction <ExternalLink class="icon icon_base" />
    </a>
  </footer>
</template>

<script setup lang="ts">
import { Heart, ExternalLink } from 'lucide-vue-next';
</script>

<style scoped>
.footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 1rem;
  padding: 1rem;
  text-align: center;
  font-size: var(--typo-size-sm);

  svg {
    display: inline;
  }
}

.footer__heart {
  color: var(--color-typo-alert);
}
</style>
