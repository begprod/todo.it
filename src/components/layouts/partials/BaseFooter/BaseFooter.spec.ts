import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import BaseFooter from '@/components/layouts/partials/BaseFooter/BaseFooter.vue';

describe('BaseFooter', () => {
  const wrapper = mount(BaseFooter);

  it('should contain css classes', () => {
    expect(wrapper.html()).toContain(
      'container flex flex-col items-center justify-center mx-auto p-5 pb-10 text-sm text-center text-gray-700',
    );
    expect(wrapper.html()).toContain('text-red-600 text-md');
    expect(wrapper.html()).toContain('underline');
    expect(wrapper.html()).toContain('inline-block ml-2 -rotate-12 origin-center');
  });

  it('should contain installation link', () => {
    expect(wrapper.html()).toContain(
      'https://github.com/begprod/todo.it#install-application-on-your-device',
    );
  });
});
