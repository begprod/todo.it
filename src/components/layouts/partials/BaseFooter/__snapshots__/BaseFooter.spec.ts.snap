// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseFooter > should match snapshot BaseFooter 1`] = `
"<footer class=\\"bg-white\\">
  <div class=\\"container flex flex-col items-center justify-center mx-auto p-5 pb-10 text-sm text-center text-gray-700\\">
    <p> made with <span class=\\"text-red-600 text-md\\">❤</span> by <a href=\\"https://github.com/begprod\\" class=\\"underline\\">begprod</a></p>
    <p><a class=\\"underline\\" href=\\"https://github.com/begprod/todo.it#install-application-on-your-device\\" target=\\"_blank\\" rel=\\"noreferrer\\"> Install app on device instruction <span class=\\"inline-block ml-2 -rotate-12 origin-center\\">→</span></a></p>
  </div>
</footer>"
`;
